nexus-and-alo/nexus-strawman.md

```diff
@@ -0,0 +1,361 @@
+
+**Nexus Task:** Unit of work which is run as a result of a request reaching the Nexus endpoint. Nexus Tasks are developed and operated similar to WorkflowTasks and ActivityTasks (eg: run and hosted on Nexus Workers). Within a Nexus Task Handler, developers can transform incoming payloads, make routing decisions and more.
+
+**Nexus Task Handler:** Source code which is designed to handle incoming Nexus Tasks. Nexus Task Handler is similar to a Workflow or Activity definition and is hosted by a Nexus Worker. 
+
+**Nexus Worker:** A special purpose Worker that pulls Nexus Tasks for its registered Nexus Instances from the Server and executes them using associated Nexus Task Handlers. 
+
+## User flows
+
+### Hosting a Nexus Service
+
+1. User instantiates Nexus Service Instance using provided `nexus` CLI commands (also possible via SDK or API) and receives an Instance ID. Nexus is now able to receive requests to its frontend endpoint (though they don't get processed until step 4).
+1. User defines a Nexus Task Handler that will handle the processing of Nexus Tasks for the instantiated Nexus Service Instance.
+1. User hosts their Nexus Task Handler using a Nexus Worker. They must specify the Nexus Service Instance ID when starting the Nexus Worker.
+
+	// TODO (Chad): show how registration would work for generic handler
+    ```go
+    func main() {
+        ctx := context.TODO()
+
+        // Create Nexus client
+        client, err := client.Dial(ctx, client.Options{Target: "nexus-backend.example.com"})
+        if err != nil {
+            log.Fatal(err)
+        }
+        defer client.Close()
+
+        // Build worker with Temporal pieces
+        temporalClient, err := temporalclient.Dial(temporalclient.Options{HostPort: "nexus-backend.example.com:7233"})
+        if err != nil {
+            log.Fatal(err)
+        }
+        defer temporalClient.Close()
+        builder, err := temporalnexus.NewWorkerBuilder(temporalnexus.WorkerBuilderOptions{Client: temporalClient})
+        if err != nil {
+            log.Fatal(err)
+        }
+        // Add workflow as ALO to Nexus
+        err = builder.AddWorkflowALOHandler("foo", temporalnexus.WorkflowALOHandlerOptions{
+            Workflow: FooWorkflow,
+            Options:  temporalclient.StartWorkflowOptions{TaskQueue: "my-task-queue"},
+        })
+        if err != nil {
+            log.Fatal(err)
+        }
+
+        // Worker definition...
+    }
+    ```
+1. Nexus is now able to process requests to its frontend endpoint.
+
+### Consuming a Nexus Service from Temporal
+
+1. User asks the Nexus Service author for the endpoint of the Nexus Service Instance they are trying to consume. For Temporal based Nexus Service Instances, the endpoint will not be HTTP or gRPC but rather a custom uri formulation.
+2. User registers provided endpoint with their SDK to avoid hardcoding the endpoint into all of their application logic. 
```

Comment:
Based on our recent discussions, this should no longer be required since Temporal server will hold the mapping.

Reply:
Will update